src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -2654,6 +2696,21 @@ def set_up_auto_upgrade_profile(self, mc: ManagedCluster) -> ManagedCluster:
             mc.auto_upgrade_profile.node_os_upgrade_channel = node_os_upgrade_channel
         return mc
 
+    def set_up_guardrails_profile(self, mc: ManagedCluster) -> ManagedCluster:
+        excludedNamespaces = self.context.get_guardrails_excluded_namespaces()
+        version = self.context.get_guardrails_version()
+        level = self.context.get_guardrails_level()
+        # provided any value?
+        if (level is not None or version is not None or excludedNamespaces is not None) and mc.guardrails_profile is None:
+            mc.guardrails_profile = self.models.GuardrailsProfile(
+                level=level,
+                version=version
+            )
+        # replace values with provided values
+        if excludedNamespaces is not None:
+            mc.guardrails_profile.excluded_namespaces = excludedNamespaces.split(",")
```

Comment:
Will the server side handle extra space after split properly?

Reply:
Changed to remove any extra spaces

Reply:
use extract_comma_separated_string instead